src/main/java/nl/tudelft/dnainator/ui/models/GraphItem.java

```diff
@@ -35,52 +42,117 @@ public GraphItem() {
 	 * @param graph	the specified graph
 	 */
 	public GraphItem(Graph graph) {
-		super(null, 0);
+		this(graph, new Group(), new Group());
+	}
+
+	/**
+	 * Construct a new top level {@link GraphItem} using the specified graph.
+	 * @param graph	the specified graph
+	 * @param content the specified graph content
+	 * @param childContent the specified child content
+	 */
+	public GraphItem(Graph graph, Group content, Group childContent) {
 		this.graph = graph;
 		this.clusters = new HashMap<>();
+		this.content = content;
+		this.childContent = childContent;
+
+		getChildren().add(content);
+		getChildren().add(childContent);
+		loadContent();
+	}
 
+	/**
+	 * Check whether this object intersects with the given viewport bounds.
+	 * @param b	the given viewport bounds
+	 * @return	true when (partially) in viewport, false otherwise
+	 */
+	public boolean isInViewport(Bounds b) {
+		return b.contains(content.localToParent(0, 0));
+	}
+
+	/*
+	 * Load the drawable content of the graph itself.
+	 */
+	private void loadContent() {
 		for (int i = 0; i < FOUR; i++) {
 			int width = NO_CLUSTERS * RANK_WIDTH / FOUR;
 			Rectangle r = new Rectangle(width, RANK_WIDTH, Color.BLACK);
 			r.setTranslateX(i * width);
-			getContent().getChildren().add(r);
+			content.getChildren().add(r);
 		}
 	}
 
-	@Override
-	public Graph getGraph() {
-		return graph;
-	}
+	/*
+	 * Load the drawable content of the graph's children, given the bounds of the viewing port.
+	 */
+	private void loadChildren(Bounds b) {
+		int minrank = (int) (Math.max(b.getMinX() / RANK_WIDTH, 0));
+		int maxrank = (int) (b.getMaxX() / RANK_WIDTH);
 
-	@Override
-	public Map<String, ClusterItem> getClusters() {
-		return clusters;
-	}
+		System.out.println("load iteration: " + minrank + " -> " + maxrank);
+		List<String> roots = graph.getRank(minrank).stream()
+				.map(e -> e.getId()).collect(Collectors.toList());
+		Map<Integer, List<Cluster>> result = graph.getAllClusters(roots, maxrank,
+				(int) (b.getWidth() / CLUSTER_DIVIDER));
+
+		clusters.clear();
+		childContent.getChildren().clear();
 
-	@Override
-	public ModelItem getRoot() {
-		return this;
+		result.forEach(this::loadRank);
+		clusters.values().forEach(this::loadEdges);
 	}
 
-	@Override
-	public void loadChildren(Bounds b) {
-		System.out.println("load iteration");
-		getChildItems().clear();
-		getChildContent().getChildren().clear();
+	/*
+	 * Load the drawable content of the edges for all displayed clusters.
+	 */
+	private void loadEdges(ClusterDrawable d) {
+		childContent.getChildren().addAll(
+				d.getClustered().stream().flatMap(e -> e.getOutgoing().stream())
+				.filter(i -> clusters.containsKey(i))
+				.filter(i -> clusters.get(i) != d)
```

Comment:
Is `d` (which could be given a better, more representative name) really the exact same `ClusterDrawable`? I.e., should `equals()` be used?